[pt][quant] Fix docs so that the example works

dskhudia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30120 [pt][quant] Fix docs so that the example works**

The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

dskhudia committed:
[pt][quant] Fix docs so that the example works

The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]

jianyuh reviewed (approved):
LGTM! Thanks for the fix!

dskhudia committed:
Update on "[pt][quant] Fix docs so that the example works"


The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]

dskhudia committed:
Update on "[pt][quant] Fix docs so that the example works"


The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]

dskhudia committed:
Update on "[pt][quant] Fix docs so that the example works"


The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]

raghuramank100 commented on torch/nn/quantized/functional.py:
```diff
@@ -94,14 +94,15 @@ def conv2d(input, weight, bias,
         >>> scale, zero_point = 1.0, 0
-        >>> dtype = torch.quint8
+        >>> dtype_inputs = torch.quint8
+        >>> dtype_filters = torch.qint8
```
Can we also add a check like what we have for padding_mode so that if dtype_inputs and dtype_filters do not match what is supported (quint8 for inputs and qint8 for filters) we error out?

dskhudia replied:
Will add

dskhudia replied:
@raghuramank100 

Added warning and also fixed the same for 3D.

dskhudia committed:
Update on "[pt][quant] Fix docs so that the example works"


The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]

dskhudia committed:
Update on "[pt][quant] Fix docs so that the example works"


The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]

dskhudia committed:
Update on "[pt][quant] Fix docs so that the example works"


The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]

dskhudia committed:
Update on "[pt][quant] Fix docs so that the example works"


The example given for functional conv2d didn't work. This diff fixes the example in docs so that it works.

Fixes https://github.com/pytorch/pytorch/issues/29649

Differential Revision: [D18604606](https://our.internmc.facebook.com/intern/diff/D18604606/)

[ghstack-poisoned]
